### ISSUE TITLE
Allow for passing additional params used in installation of tools

### ISSUE THREAD

**Issue by pgt502** [twitchtv/retool#55]
Currently, each tool is installed using `go install` command which is pretty limiting. Certain tools require arguments to be passed in when building them. In order to accommodate that, the `go build` command needs to be used.

## Solution
Allow for passing extra parameters when building each tool which can be defined in the tools.json.

For example:
```json
{
      "Repository": "github.com/golang-migrate/migrate/cli",
      "Commit": "4937cd088f7c7193ee59b319c1c2caa7482aae8e",
      "ExecutableName": "migrate",
      "PreParams": [
          "-tags", 
          "'postgres'"
      ]
    }
```

In this case, PreParams are the arguments passed in just after `go build` command and before the path and other arguments.

**Comment by spenczar** [comment]
Can we perhaps narrow this to only being about build tags, rather than arbitrary arguments? That's a lot simpler to reason about.

**Comment by pgt502** [comment]
Yes, I agree it would make it much simpler, however, if it is likely that other parameters will be needed in the future, it might make sense to implement it more generically. What do you think?

**Comment by WIZARDISHUNGRY** [comment]
Came here for this exact utility and tag
